### ISSUE TITLE
Cache failed loads to prevent reoccurring error logs

### ISSUE THREAD

**Issue by willnationsdev** [willnationsdev/godot-inheritance-dock#16]
Currently, if you have ANY script in your project (or scene / resource for that matter) which fails to load, the Inheritance Dock will continue to attempt to load everything all over again every time the filesystem for the project is changed (so, if you save or move any file basically). If you are updating old code or have en mass rendered several files out of date (updated the engine version perhaps?), then a whole bunch of assets will become filled with errors all at once, and you'll get flooded with an error log for each resource, every time a change is made.

This can be fixed by keeping a dictionary of which resources failed to load, only attempting to load ones which have not yet failed to load, and finally, using the `resource_saved` signal in the 3.1 release, flagging which ones are ready for another load attempt.

**Comment by chanon** [comment]
I would like to use your extension, but this is making every file save very slow.
I get a bunch of errors logged, but I have no idea which scenes are causing them, so I would like to suggest logging the scenes/resources that caused the errors too.

**Comment by willnationsdev** [comment]
Yeah, I agree, it can definitely stand to improve. The error report change is a good idea. I'll try to get these changes done when I can, but it may be a little while. Thanks for the feedback!

**Comment by willnationsdev** [comment]
I can go ahead and modify the error report sooner rather than later, but the rest of it will have to wait for 3.1 unfortunately. 

Edit: actually, I'll have to check and see whether my resource\_saved signal was cherry-picked into a 3.0.x build. 

Edit: lol, nope. Still hasn't even been integrated into the master branch.

**Comment by willnationsdev** [comment]
Also, fyi, I'm working on some engine changes that should allow me to offload a lot of these cache-updates and inheritance checking to the C++ code which will also speed up the plugin's dock-reloads immensely.
